okhttp-tls/src/main/java/okhttp3/tls/HandshakeCertificates.kt

```diff
@@ -67,20 +67,49 @@ import javax.net.ssl.X509TrustManager
  *    roots private to an organization or service.
  */
 class HandshakeCertificates private constructor(
-  private val keyManager: X509KeyManager,
-  private val trustManager: X509TrustManager
+  @get:JvmName("keyManager") val keyManager: X509KeyManager,
+  @get:JvmName("trustManager") val trustManager: X509TrustManager
 ) {
+
+  @JvmName("-deprecated_keyManager")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "keyManager"),
+      level = DeprecationLevel.WARNING)
   fun keyManager(): X509KeyManager = keyManager
 
+  @JvmName("-deprecated_trustManager")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "trustManager"),
+      level = DeprecationLevel.WARNING)
   fun trustManager(): X509TrustManager = trustManager
 
-  fun sslSocketFactory(): SSLSocketFactory = sslContext().socketFactory
+  @get:JvmName("sslSocketFactory") val sslSocketFactory: SSLSocketFactory
+    get() {
+      return sslContext.socketFactory
+    }
+
+  @JvmName("-deprecated_sslSocketFactory")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "sslSocketFactory"),
+      level = DeprecationLevel.WARNING)
+  fun sslSocketFactory(): SSLSocketFactory = sslSocketFactory
 
-  fun sslContext(): SSLContext {
-    return Platform.get().newSSLContext().apply {
-      init(arrayOf<KeyManager>(keyManager), arrayOf<TrustManager>(trustManager), SecureRandom())
+  @get:JvmName("sslContext") val sslContext: SSLContext
+    get() {
+      return Platform.get().newSSLContext().apply {
```

Comment:
I'd argue against this being a val

Reply:
Hmmm . . . I think SSLContext is immutable, which feels like a `val` to me.

Reply:
But it makes it look externally cheap, while internally it's creating this expensive (potentially) native implement SSLContext and calling init on it.  It's the kind of thing that Intellij would suggest extracting a common variable and reusing, but calling once vs calling 10 times is very different.  The equality of two SSLContext isn't known IIUC.  And they may be immutable, but there is state in SSLContext potentially like session ids etc.

Reply:
Works for me.

Reply:
Thanks for your feedback! I removed the wrong `@get:JvmName(...) val` for `sslContext` and `sslSocketFactory`.